numba/np/arrayobj.py

```diff
@@ -1633,8 +1672,13 @@ def fancy_setslice(context, builder, sig, args, index_types, indices):
     indexer = FancyIndexer(context, builder, aryty, ary,
                            index_types, indices)
     indexer.prepare()
+    src_type = None
 
     if isinstance(srcty, types.Buffer):
+        if not context.enable_nrt:
+            raise NotImplementedError("This type of indexing is not currently"
+                                      " supported for given compiler target.")
```

Comment:
As above, ideally this would be tested, but I'm inclined to leave this to a subsequent PR as I am relatively sure it's going to need to be moved.